app/services/preserved_object_handler.rb

```diff
@@ -92,14 +140,12 @@ def update_per_version_comparison(db_object)
   def update_db_object(db_object, results)
     if db_object.changed?
       db_object.save
-      results << result_hash(UPDATED_DB_OBJECT)
+      results << result_hash(UPDATED_DB_OBJECT, db_object.class.name)
     else
       # FIXME: we may not want to do this, but instead to update specific timestamp for check
       db_object.touch
-      results << result_hash(UPDATED_DB_OBJECT_TIMESTAMP_ONLY)
+      results << result_hash(UPDATED_DB_OBJECT_TIMESTAMP_ONLY, db_object.class.name)
     end
-  rescue ActiveRecord::ActiveRecordError => e
-    results << result_hash(DB_UPDATE_FAILED, "#{e.inspect} #{e.message} #{e.backtrace.inspect}")
```

Comment:
I see that rescue ActiveRecordError was moved up the method chain -- I'm fine with it but wonder if there's a principle or a "teach naomi" moment here?

Reply:
i think we wanted the catch to apply to the entire operation of creating the `PreservedObject` and the `PreservationCopy`.  if the former fails, we don't want to move on to the latter, but as it was, an error in pres copy creation would've been trapped, and would've allowed processing to move onto something that depends on a thing we just failed to create.

we may also want to wrap that whole thing in an explicit transaction... i think declaring transactions isn't something we've thought about much so far, but we might at least want to file a tech debt ticket to figure out where (if anyplace) we might need them.

Reply:
I'm clueless about how to do transactions, but I think this makes sense.